src/shared/reboot-util.c

```diff
@@ -36,6 +39,22 @@ int update_reboot_parameter_and_warn(const char *parameter) {
         return 0;
 }
 
+int get_reboot_parameter(char **parameter) {
+        int r;
+
+        assert(parameter);
+
+        *parameter = NULL;
+
+        r = read_one_line_file("/run/systemd/reboot-param", parameter);
+        if (r < 0 && r != -ENOENT) {
+                log_debug_errno(r, "Failed to read /run/systemd/reboot-param: %m");
+                return r;
```

Comment:
log_debug_errno() returns the error you pass in, hence you can merge the above lines into:

```c
return log_debug_errno(r, …
```

Reply:
Changed as suggested